pkg/emulator/main.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go-emulator/emulator"
+	"math"
+	"os"
+	"strings"
+)
+
+func main() {
```

Comment:
Then this should be named `TestEmulator(t *testing.T)`.

You can use the `require` package to make assertions within the test.